tools/replicate_server_settings.rb

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env ruby
+require File.expand_path("../config/environment", __dir__)
+require 'trollop'
+
+opts = Trollop.options(ARGV) do
+  banner "USAGE:   #{__FILE__} -s <server id> -p <settings path separated by a /> \n" \
+         "Example: #{__FILE__} -d -s 1 -p ems/ems_amazon/additiona_instance_types"
+
+  opt :dry_run,  "Dry Run",                                                :short => "d"
+  opt :serverid, "Replicating source server Id (default: current server)", :short => "s"
+  opt :path,     "Replicating source path within advanced settings hash",  :short => "p", :default => ""
+end
+
+puts opts.inspect
+
+Trollop.die :path, "is required" unless opts[:path_given]
+
+server = opts[:serverid] ? MiqServer.find(opts[:serverid]) : MiqServer.my_server
+settings = server.get_config("vmdb")
+
+config = settings.config
+path = opts[:path].split("/").map(&:to_sym)
+values = config.fetch_path(path)
+target_servers = MiqServer.where.not(:id => server.id)
+
+puts "Replicating from server id=#{server.id}, path=#{opts[:path]} to #{target_servers.count} servers"
+target_servers.each do |target|
+  puts " - replicating to server id=#{target.id}..."
+  target_settings = target.get_config("vmdb")
+  target_settings.config.store_path(*path, values)
+
+  valid, errors = VMDB::Config::Validator.new(target_settings).validate
```

Comment:
You shouldn't need to do this, `add_settings_for_resource` calls `Vmdb::Settings.save!` which [does validation for you](https://github.com/ManageIQ/manageiq/blob/07decc471b743fc04421d0891898dd9166521021/lib/vmdb/settings.rb#L50), plus I don't see invalid config being much of an issue as the settings you're saving have already been saved for a different server :wink:

Reply:
great! Done